examples/two-columns-layout/src/app.tsx

```diff
@@ -1,18 +1,40 @@
 /** @jsx h */
 import { autocomplete } from '@algolia/autocomplete-js';
 import { h, render } from 'preact';
+import { pipe } from 'ramda';
 
+import { populate, uniqBy } from './functions';
 import { articlesPlugin } from './plugins/articlesPlugin';
 import { brandsPlugin } from './plugins/brandsPlugin';
 import { categoriesPlugin } from './plugins/categoriesPlugin';
 import { faqPlugin } from './plugins/faqPlugin';
 import { popularPlugin } from './plugins/popularPlugin';
 import { productsPlugin } from './plugins/productsPlugin';
 import { querySuggestionsPlugin } from './plugins/querySuggestionsPlugin';
+import { quickAccessPlugin } from './plugins/quickAccessPlugin';
 import { recentSearchesPlugin } from './plugins/recentSearchesPlugin';
 
 import '@algolia/autocomplete-theme-classic';
 
+const removeDuplicates = uniqBy(({ source, item }) => {
+  if (
+    ['recentSearchesPlugin', 'querySuggestionsPlugin'].indexOf(
+      source.sourceId
+    ) === -1
+  ) {
+    return item;
+  }
+
+  return source.sourceId === 'querySuggestionsPlugin' ? item.query : item.label;
+});
+
+const fillWith = populate({
+  mainSourceId: 'querySuggestionsPlugin',
+  limit: 8,
+});
```

Comment:
This is hard to understand, especially without knowing exactly what `populate` does.

Reply:
I added some comments in the `populate` file. I hope that's cleared now, also I'd like to find a better function names (`populate` and `fillWith`) and parameter names (`mainSourceId` and `limit`) cause I'm not sure they are very clear right now, WDYT?